### DISCUSSION TITLE
Use with petite-vue

### DISCUSSION THREAD

**Discussion by arnoson** [ElMassimo/iles#27] [Ideas]
Really promising project, exactly what I've been looking for :)
Would it also be possible to use petite-vue instead of vue in the generated website?

**Comment by ElMassimo** [comment]
Hi @arnoson! At the moment, there is no library that can compile standard vue components to Vue, so currently there's no way to do that transparently.

In îles you can use [plain JS components](https://iles-docs.netlify.app/guide/client-scripts#vanilla-components) where you would be able to import `petite-vue` or any library, and use it as needed. You can also run a script that attaches itself to [Vue-rendered HTML](https://iles-docs.netlify.app/guide/client-scripts#client-script-block).

If you are looking to have a smaller runtime but still prerender the components, I'd recommend using [preact or solid](https://iles-docs.netlify.app/config/plugins#vite-plugin-solid) islands.
